Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/viewholder/AchievementProgressViewHolder.kt

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.features.settings.ui.viewholder
+
+import android.view.View
+import android.widget.CompoundButton
+import androidx.lifecycle.ViewModelProvider
+import org.dolphinemu.dolphinemu.databinding.ListItemAchievementProgressBinding
+import org.dolphinemu.dolphinemu.features.settings.model.AchievementProgressViewModel
+import org.dolphinemu.dolphinemu.features.settings.model.SingleAchievement
+import org.dolphinemu.dolphinemu.features.settings.ui.AchievementProgressItem
+import org.dolphinemu.dolphinemu.features.settings.ui.SettingsActivity
+
+class AchievementProgressViewHolder(private val binding: ListItemAchievementProgressBinding) :
+    AchievementProgressItemViewHolder(binding.getRoot()),
+    View.OnClickListener,
+    CompoundButton.OnCheckedChangeListener {
+    private lateinit var viewModel: AchievementProgressViewModel
+    private lateinit var achievement: SingleAchievement
+    private var position = 0
+
+    override fun bind(activity: SettingsActivity, item: AchievementProgressItem, position: Int) {
+        viewModel = ViewModelProvider(activity)[AchievementProgressViewModel::class.java]
+        achievement = item.achievement!!
+        this.position = position
+        binding.achievementTitle.text = achievement.title
+        binding.achievementDescription.text = achievement.description
+        binding.achievementScore.text = achievement.points.toString()
+        binding.achievementStatus.text = if (achievement.unlocked == 0) "Locked" else "Unlocked"
+        binding.achievementProgress.text = achievement.measuredProgress
+    }
+
+    override fun onClick(root: View) {
+    }
+
+    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
```

Comment:
There's no reason to implement this or the `CompoundButton.OnCheckedChangeListener` interface.